app/controllers/application_controller.rb

```diff
@@ -52,6 +52,10 @@ class ApplicationController < ActionController::Base
 
   rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
 
+  rescue_from Rack::Timeout::RequestTimeoutException do
+    redirect_to timeout_path
```

Comment:
Instead of triggering another request, could we render the timeout template here?